internal/docker/builder.go

```diff
@@ -426,26 +457,39 @@ func deployImage(
 	}
 
 	if os.Getenv("COMPLEMENT_CA") == "true" {
-		caVolume, caMount, err = getCaVolume(docker, ctx)
+		var caMount mount.Mount
+		caMount, err = getCaVolume(ctx, docker)
 		if err != nil {
 			return nil, err
 		}
+
+		mounts = append(mounts, caMount)
+	}
+
+	asMount, err := getAppServiceVolume(ctx, docker)
+	if err != nil {
+		return nil, err
+	}
+	mounts = append(mounts, asMount)
+
+	env := []string{
+		"SERVER_NAME=" + hsName,
+		"COMPLEMENT_CA=" + os.Getenv("COMPLEMENT_CA"),
 	}
 
 	body, err := docker.ContainerCreate(ctx, &container.Config{
 		Image: imageID,
-		Env:   []string{"SERVER_NAME=" + hsName, "COMPLEMENT_CA=" + os.Getenv("COMPLEMENT_CA")},
+		Env:   env,
 		//Cmd:   d.ImageArgs,
 		Labels: map[string]string{
 			complementLabel:        contextStr,
 			"complement_blueprint": blueprintName,
 			"complement_hs_name":   hsName,
 		},
-		Volumes: caVolume,
```

Comment:
Removing all of the `Volumes` config as it doesn't seem necessary and works without it. We use the `container.HostConfig,Mounts` already which covers this use case.

 - Seems like the `Volumes` syntax is used to create an anonymous volume, https://stackoverflow.com/a/58916037/796832
 - And lots of people not knowing what `Volumes` syntax is or what to do. Seems like Mounts is the thing to use
    - https://github.com/fsouza/go-dockerclient/issues/155
    - https://stackoverflow.com/questions/55718603/golang-docker-library-mounting-host-directory-volumes
    - https://stackoverflow.com/questions/48470194/defining-a-mount-point-for-volumes-in-golang-docker-sdk